### PR TITLE
Use Meta.WaylandClient for shell components

### PR DESCRIPTION
**Pull request by leolost2605**
Tries to fix stuff like #1693 and #1694 by using `Meta.WaylandClient` to launch the respective apps as trusted subprocesses and by managing their position etc. from within gala as [suggested](https://github.com/elementary/gala/issues/1693#issuecomment-1593531836) by @davidmhewitt.
As a proof of concept this PR currently positions the new dock in the top left corner (I didn't bother getting the maths right for the correct position with the possibility we might not even use this :)) of the primary monitor, keeps it above all other windows and hides it from the window list.
It's based on the `wayland-testing` branch because for some reason `master` still crashes for me but can be easily rebased.
@tintou @Marukesu is this worth pursuing?

The code is probably not very good, it's the first time I do something window manager related and this is just the result of me playing around with wayland :) Still it works and shouldn't require too much additional effort to get it ready 🤞

### CODE REVIEW

File: src/PantheonShell.vala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        display.window_created.connect ((window) => {
+            if (dock_client != null && dock_client.owns_window (window)) {
+                setup_dock_window (window);
+            }
+        });
+
+        var subprocess_launcher = new GLib.SubprocessLauncher (NONE);
+        try {
+            dock_client = new Meta.WaylandClient (subprocess_launcher);
+            string[] args = {"io.elementary.dock"};
```
Comment (leolost2605) [inline src/PantheonShell.vala:26]: For a more useful way to test this replace dock with wingpanel and remove the TryExec for io.elementary.wingpanel from the wayland session settings and it works really fine apart from maximized windows disappearing behind the panel :)

**Comment by tintou** [comment]
Hi there, you should take a look at https://github.com/elementary/gala/pull/1705

**Comment by leolost2605** [comment]
Hi, yes I've seen that and even tried to do some direct xdg shell wayland access from a client side to see how easy/hard using custom wayland protocols is but at some point (I think when I just couldn't figure out how to use the headers generated for wayland protocols from within vala without manually writing vapi, although I can imagine I probably did something very wrong🙂) just gave up and found this to be at least IMHO a much simpler solution that (as @davidmhewitt pointed out) doesn't require new API and any (?) client side effort

**Comment by leolost2605** [comment]
Also with this it's easier for new people, like me, to contribute without having to dig deep down into how wayland works :)

**Pull request closed without merging**